### DISCUSSION TITLE
Post - Mastering Key Remapping on Linux with xremap: A Survival Guide

### DISCUSSION THREAD

**Discussion by paolomainardi** [paolomainardi/paolomainardi.com#34] [General]
Comments for article: https://www.paolomainardi.com/posts/linux-remapping-keys-with-xremap/

**Comment by hced** [comment]
Nice article. Got me up and running. However I got stuck enabling the systemd service. Any clues why? On Arch, using Gnome and Hyprland on and off (new to Linux):

`Failed to enable unit: Invalid unit name multi-user.targets`

EDIT: Oh, I think there's a typo there; should probably be `target` in singular

**Comment by janekmi** [comment]
Thank you very much for this article. It helped me a lot.

One thing which is missing in my opinion is that xremap requires priviledges to work. It seems not sufficiently explained how your user-level service was able to provide them. I end up making it an /etc/systemd/system-level service. I would love to see how you solved this problem.

Other then that, a really great article. I love especially the theoretical introduction. It gave me the confidence xremap does the thing as it should.

Thanks!

**Comment by paolomainardi** [comment]
Dear @hced and @janekmi, to run the systemd unit as a user, xremap must be capable of running without sudo. You can follow these instructions:  https://github.com/xremap/xremap?tab=readme-ov-file#running-xremap-without-sudo
